pints/_abc/_abc_smc.py

```diff
@@ -0,0 +1,232 @@
+#
+# ABC SMC method
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import pints
+import numpy as np
+
+
+class ABCSMC(pints.ABCSampler):
+    r"""
+    Implements the ABC-SMC algorithm as described in [1]. This
+    algorithm is also referred to as ABC Population Monte Carlo
+    (ABC PMC) [2].
+
+    In this algorithm we try to obtain ``nr_samples`` samples at
+    once. In order to transition between each set of ``nr_samples``,
```

Comment:
This can easily be misunderstood as "This algorithm is just sample(nr_samples)". 

You also don't need to introduce `nr_samples` in the very first line. First tell me that there are going to be several rounds of ABC, each with a different proposal distribution (the SMC bit). Only once that's clear go into sizes of intermediate distributions etc.

Reply:
You could even start with an example?

> In ABC methods, typically, a distance metric comparing the observed data and the simulated is used. Here, we use the root mean square error (RMSE), and we accept a parameter value if the for the first intermediate distribution, for the second one and for the last. 

This bit from the notebook is much clearer than the docstring!